pandas/plotting/_core.py

```diff
@@ -2951,21 +2951,46 @@ def box(self, by=None, **kwds):
 
     def hist(self, by=None, bins=10, **kwds):
         """
-        Histogram
+        Draw one histogram of the DataFrame's columns.
+
+        A histogram is a representation of the distribution of data.
+        This function groups the values of all given Series in the DataFrame
+        into bins, and draws all bins in only one matplotlib.AxesSubplot. This
+        is useful when the DataFrame's Series are in a similar scale.
 
         Parameters
         ----------
-        by : string or sequence
+        by : str or sequence, optional
             Column in the DataFrame to group by.
-        bins: integer, default 10
-            Number of histogram bins to be used
-        `**kwds` : optional
-            Additional keyword arguments are documented in
-            :meth:`pandas.DataFrame.plot`.
+        bins : int, default 10
+            Number of histogram bins to be used.
+        **kwds : optional
+            Keyword arguments to pass on to :meth:`pandas.DataFrame.plot`.
 
         Returns
         -------
-        axes : :class:`matplotlib.axes.Axes` or numpy.ndarray of them
+        axes : matplotlib.AxesSubplot histogram.
+
+        See Also
+        --------
+        pandas.DataFrame.hist : Draw histograms per DataFrame's Series.
+        pandas.Series.hist : Draw a histogram with Series' data.
+
+        Examples
+        --------
+        When using values between 0 and 3, calling hist() with bins = 3 will
+        create three bins: one that groups values between 0 and 1, another
+        for values between 1 and 2, and another for values between 2 and 3.
+        We use alpha parameter to be able to see overlapping columns.
+
+        .. plot::
+            :context: close-figs
+
+            >>> df = pd.DataFrame({
+            ...     'length': [ 1.5, 0.5, 1.2, 0.9, 3],
+            ...     'width': [ 0.7, 0.2, 0.15, 0.2,  1.1]
+            ...     }, index = ['pig', 'rabbit', 'duck', 'chicken', 'horse'])
+            >>> hist = df.plot.hist(bins = 3, xticks = range(4), alpha = 0.5)
```

Comment:
Although I like the fact you used some more "real world" data, the problem is that with only 5 values, the histrogram will look rather strange. So I would maybe still generate some more random data in this case.

Reply:
During the sprint some of us actually discussed about it: if you generate random data, each time the histogram will be different, and it will be kind of difficult to understand and to explain. So I tried to create some "real" data that can be small enough to generate a clear histogram that shows how it works. But if you think this data is not enough, I could try to improve it.

Reply:
If you want reproducible data, you can set a random seed. Also, if you use quite some data eg from a normal distrubution, the plot will look very similar each time.

I plotted the data above, and indeed I think it is too small, IMO it gives not a very good "sense" of what a good/useful histogram looks like.

Reply:
+1 for using random data, as it imparts when histograms are useful (summarizing the distribution of a larger dataset).

Reply:
I have been thinking about it but I can't imagine an easy way to generate random data that, at the same time, it's __easy to understand__ in a compact form (this is just an Example section) and also shows the __direct relation__ with the resulting histogram. Could you tell me any suggestion?

On the other hand, in pandas.DataFrame.hist() docstring they used another "toy" example (which actually was my work too) and it's already merged...

Reply:
```
In [64]: df = pd.DataFrame({'a': np.random.randn(1000), 'b': np.random.randn(1000)*1.5 + 2})

In [65]: df.plot.hist(alpha=0.5)
Out[65]: <matplotlib.axes._subplots.AxesSubplot at 0x7fbaadcea6d8>
```

gives me

![figure_1-1](https://user-images.githubusercontent.com/1020496/37592753-f7cca98a-2b6f-11e8-9244-7f6360fc834d.png)

which I think is a reasonable plot to show functionality of hist ?

> On the other hand, in pandas.DataFrame.hist() docstring they used another "toy" example (which actually was my work too) and it's already merged...

That's true, but IMO we should change it there as well.

Reply:
Ok, I found a random example that sounds quite natural and understandable. I hope you'll like it.